README.md

```diff
@@ -102,7 +102,55 @@ it allows to initialize 'sql_duration' map entry to 0 only if this map entry is
 
 ## Example #3
 
-Third use case : you have no specific start event and no specific end event.  
+Third use case: You have a start event, however no specific end event. 
+
+A typical case is aggregating or tracking user behaviour. We can track a user by its ID through the events, however once the user stops interacting, the events stop coming in. There is no specific event indicating the end of the user's interaction.
+
+In this case, we can enable the option 'push_map_as_event_on_timeout' to enable pushing the aggregation map as a new event. 
+In addition, we can enable 'timeout_code' to execute code on the populated timeout event.
+We can also add 'timeout_task_id_field' so we can correlate the task_id, which in this case would be the user's ID. 
+
+```
+    INFO - 12345 - Clicked One
+    INFO - 12345 - Clicked Two
+    INFI - 12345 - Clicked Three
+```
+
+* You can aggregate the amount of clicks the user did like this:
+
+``` ruby
+    filter {
+        grok {
+                 match => [ "message", "%{LOGLEVEL:loglevel} - %{NOTSPACE:taskid} - %{GREEDYDATA:msgtext}" ]
+        }
+
+        aggregate {
+            task_id => "%{taskid}"
+            code => "map['clicks'] ||= 0; map['clicks'] += 1;"
+            timeout_task_id_field => "userId"
```

Comment:
In logstash/ruby, naming convention is more either "userid", either "user_id".

That said, being a java developer (like you), I understand why you used camel case :)